src/pages/Register/index.tsx

```diff
@@ -40,18 +40,15 @@ const Register = () => {
       setMessage('Passwords do not match')
     }
 
-    if (role === jobseeker) {
-      if (password === confirmPassword) {
+    if (password === confirmPassword) {
+      if (role === jobseeker) {
         dispatch(registerJobseekerRequest(email, password))
-        setMessage('Registered successfully')
-        setTimeout(() => history.push('/login'), 2000) // to make confirmation of message box of registration
       }
-    } else if (role === employer) {
-      if (password === confirmPassword) {
+      if (role === employer) {
         dispatch(registerEmployerRequest(email, password))
-        setMessage('Registered successfully')
-        setTimeout(() => history.push('/login'), 2000)
       }
+      setMessage('Registered successfully')
+      setTimeout(() => history.push('/login'), 2000)
```

Comment:
I think it's better if you could redirect right inside the saga, when a success response has come back. You could pass the `history` object to the `register` action, extract this object inside the saga, then call `push` on it when a 200 response is received.